### PR TITLE
update scancode for version 3.x

### PR DESCRIPTION
**Pull request by dabutvin**
adds a test setup for whole scancode output

### CODE REVIEW

**Comment by dabutvin** [comment]
image

**Comment by dabutvin** [comment]
@pombredanne does this change make sense to you?

Is `declared_licensing` reading from license in package.json?

File: providers/summary/scancode.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ class ScanCodeSummarizer {
 
   _summarizePackageInfo(files) {
     for (let file of files) {
-      const asserted = get(file, 'packages[0].asserted_licenses')
+      const asserted = get(file, 'packages[0].asserted_licenses') || get(file, 'packages[0].declared_licensing')
```
Comment (dabutvin) [inline providers/summary/scancode.js]: i'm seeing this is actually `declared_license` not `declared_licensing`

File: providers/summary/scancode.js [hunk 2]
```diff
@@ -58,16 +59,11 @@ class ScanCodeSummarizer {
...
-      if (asserted) {
-        const packageLicenses = addArrayToSet(
-          asserted,
-          new Set(),
-          license => license.license || license.spdx_license_key
-        )
-        return this._joinExpressions(packageLicenses)
-      }
+      const declared =
+        SPDX.normalize(get(file, 'packages[0].declared_license') || get(file, 'packages[0].declared_licensing')) ||
```
Comment (jeffmcaffer) [inline providers/summary/scancode.js]: check whether we need to map scancode license ids to SPDX ids. consider creating a `scancodeNormalize` function

File: providers/summary/scancode.js [hunk 3]
```diff
@@ -99,6 +86,26 @@ class ScanCodeSummarizer {
...
+  }
+
+  _parseFileLicenses(fileLicenses) {
+    if (!fileLicenses) return null
+    let licenses = new Set(fileLicenses.map(x => x.license).filter(x => x))
+    if (!licenses.size) {
+      licenses = new Set(
+        fileLicenses
+          .filter(x => x.score >= 80)
+          .map(license => this._createExpressionFromRule(license.matched_rule, license.spdx_license_key))
```
Comment (jeffmcaffer) [inline providers/summary/scancode.js]: fix typo `scanodeMap` in `_createExpressionFromRule`

File: test/fixtures/scancode/2.9.8/npm-basic.json [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+        "is_archive": false,
+        "is_media": false,
+        "is_source": false,
+        "is_script": false,
+        "licenses": [],
+        "license_expressions": [],
+        "holders": [],
+        "copyrights": [],
+        "authors": [],
+        "packages": [
```
[the remark this answers is not in the document]
Comment (dabutvin) [reply test/fixtures/scancode/2.9.8/npm-basic.json]: sorry I lost this comment. I can confirm, this is data output from running the crawler image.
I am adding 3.0.0 data now

**Pull request merged**